packages/rum/src/domain/record/utils.ts

```diff
@@ -49,3 +50,22 @@ export function getPathToNestedCSSRule(rule: CSSRule): number[] | undefined {
 
   return path
 }
+
+export type ListenerHandler = () => void
+
+export function getEventTarget(event: Event): Node {
+  if (event.composed === true && isNodeShadowHost(event.target as Node)) {
+    return event.composedPath()[0] as Node
+  }
+  return event.target as Node
+}
+
+const recordIds = new WeakMap<Event, number>()
+let nextId = 1
+
+export function getRecordIdForEvent(event: Event): number {
+  if (!recordIds.has(event)) {
+    recordIds.set(event, nextId++)
+  }
+  return recordIds.get(event)!
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏would it worth to have a `observers/utils` file ?

Reply:
I'd say it is a balance between having one file by utility code and all utility code in one files.
Since these utilities has been moved to `record/utils`, we loose the information that it is used only by `record/observers` files. 
Having `record/observers/utils` could make that clearer and would provide some kind of encapsulation of this code inside (if needed).